lib/pure/uri.nim

```diff
@@ -147,6 +151,7 @@ proc encodeQuery*(query: openArray[(string, string)], usePlus = true,
     if not omitEq or val.len > 0:
       result.add('=')
       result.add(encodeUrl(val, usePlus))
+  if usePrefix and result.len > 0: result = "?" & result
```

Comment:
So the implementation is just as inefficient as doing `` "?" & encodeQuery(x)``? Quite an achievement. :P